document-policy-explainer.md

```diff
@@ -0,0 +1,337 @@
+<iframe policy="no-scripts">
+<iframe sandbox policy="scripts"
+<iframe sandbox="no-scripts" policy="scripts">
+<iframe policy="no-same-origin">
+```
+
+None of these result in a `Sec-Required-Feature-Policy` header, as all of the
+existing sandbox features are safelisted.
+
+However, new features are not safelisted, and so would result in a header if any
+of them are restricted:
+
+```html
+<iframe policy="no-scripts, no-document-write">
+```
+
+results in a request header:
+
+```http
+Sec-Required-Document-Policy: no-document-write
+```
+
+(Note that `no-scripts` is not present in the header; only those values which
+are required to appear in the response header are listed.)
+
+The sandbox attribute in an iframe tag serves to disable all of the
+currently-defined sandbox features. These can also be toggled individually with
+the policy attribute:
+
+```html
+<iframe policy="no-document-write, no-escape-in-popups">
+```
+
+They can also be set from an HTTP header:
+
+```http
+Require-Document-Policy: no-forms
+```
+
+Would impose the sandboxed forms browsing context flag on all nested content.
+
+### Sandboxing yourself
+
+```http
+Document-Policy: no-popups, no-modals, no-presentation-lock, no-forms,
+                 no-pointer-lock
+```
+
+In this case (like in the first document policy example), the document may embed
+other content without these restrictions.
+
+You can also include both headers, to sandbox yourself as well as all nested
+content:
+
+```http
+Document-Policy: no-popups, no-modals, no-presentation-lock, no-forms, no-pointer-lock
+Require-Document-Policy: no-popups, no-modals, no-presentation-lock, no-forms, no-pointer-lock
+```
+
+## So, how does it work?
+
+`Document-Policy` is an HTTP response header that sets the policy on the
+document which it is served with. You can use it to set things like restrictions
+on image sizes or compression ratios, lazy frame and image loading policies, use
+of synchronous APIs, or use of APIs that fall under iframe sandboxing,
+traditionally.
+
+That header doesn't impose any restrictions on content which the document embeds
+at all, and in the base case, every document is free to set its own policy
+independently.
+
+(In a minute, we'll talk about required policies, though -- and not conforming
+to a required policy might impact your page's ability to be loaded, but in
+general, you're free to set your policy any way you like)
+
+Policies controllable with this mechanism include things like document
+configuration, and sandboxing. Each of them can be named with a unique string.
+
+Examples:
+
+* `image-compression`
+* `frame-loading`
+* `vertical-scroll`
+* `popups`
+* `document-write`
+* `scripts`
+
+Some policies are boolean switches, and can be turned off with the syntax
+"`no-<feature-name>`":
+
+* `no-popups`
+* `no-scripts`
+* `no-document-write`
+
+Others take parameters, whose types and names are specific to each policy:
+
+* `image-compression;min-bpp=2`
+* `frame-loading:lazy`
```

Comment:
Should this be a semicolon instead of a colon?

Reply:
Yes, thanks!